presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java

```diff
@@ -54,28 +57,37 @@ public WindowPartition(PagesIndex pagesIndex,
             int partitionStart,
             int partitionEnd,
             int[] outputChannels,
-            List<WindowFunction> windowFunctions,
-            FrameInfo frameInfo,
+            List<FramedWindowFunction> windowFunctions,
             PagesHashStrategy peerGroupHashStrategy)
     {
         this.pagesIndex = pagesIndex;
         this.partitionStart = partitionStart;
         this.partitionEnd = partitionEnd;
         this.outputChannels = outputChannels;
         this.windowFunctions = windowFunctions;
-        this.frameInfo = frameInfo;
+        this.frameInfo = assertSingleFrame(windowFunctions); // TODO remove this when multiple Frames are handled correctly
         this.peerGroupHashStrategy = peerGroupHashStrategy;
 
         // reset functions for new partition
         WindowIndex windowIndex = new PagesWindowIndex(pagesIndex, partitionStart, partitionEnd);
-        for (WindowFunction windowFunction : windowFunctions) {
-            windowFunction.reset(windowIndex);
+        for (FramedWindowFunction framedWindowFunction : windowFunctions) {
+            framedWindowFunction.function.reset(windowIndex);
         }
 
         currentPosition = partitionStart;
         updatePeerGroup();
     }
 
+    private static FrameInfo assertSingleFrame(List<FramedWindowFunction> windows)
+    {
+        if (windows.isEmpty()) {
+            return new FrameInfo(WindowFrame.Type.RANGE, UNBOUNDED_PRECEDING, Optional.empty(), CURRENT_ROW, Optional.empty());
```

Comment:
This was actually a bug (but didn't want to change logic in refactor only). It's already fixed in patch by @ebd2 (here: https://github.com/prestodb/presto/pull/5860).
Fixup removing this check (since it's obsolete after above's merged) is here: 54492241183d00b614f477ccebb42dc3c277bad2